receiver/elasticsearchreceiver/metadata.yaml

```diff
@@ -779,3 +779,57 @@ metrics:
       value_type: int
     attributes: [operation, index_aggregation_type]
     enabled: true
+  elasticsearch.index.segments.count:
+    description: Number of segments of an index.
+    unit: "{segments}"
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
+  elasticsearch.index.segments.size:
+    description: Size of segments of an index.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
+  elasticsearch.index.segments.doc_values.size:
+    description: Size of memory for doc_values of an index.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
+  elasticsearch.index.segments.fixed_bit_set.size:
+    description: Size of memory for fixed BitSet of an index.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
+  elasticsearch.index.segments.index_writer.size:
+    description: Size of memory for index writer of an index.
+    unit: By
+    sum:
+      monotonic: false
+      aggregation: cumulative
+      value_type: int
+    attributes: [index_aggregation_type]
+    enabled: true
+  elasticsearch.index.segments.terms.size:
```

Comment:
Since all of these are measurements of memory, should the namespace be `elasticsearch.index.segments.memory`?

Reply:
Makes sense. Should then these be merged into one metric and be differentiated by an attribute?

Reply:
> Makes sense. Should then these be merged into one metric and be differentiated by an attribute?

That's a good question. The answer depends on whether or not the measurements would be overlapping. If we know for sure they are _not_ overlapping, then it makes sense to aggregate them together. Otherwise, the "total" that we would get from aggregating them together would not be meaningful. 

It's also possible that some of the metrics could be combined according to these guidelines, but others are kept separate. I'm not familiar enough with these metrics to say one way or the other, but as an example, it _might_ be the case that `elasticsearch.index.segments.size` is a broad measurement that overlaps some of the others so would be kept separate, but maybe the others are mutually exclusive of each other so could be combined.

Reply:
Unfortunately, the docs don't help much, but after some research I think that these statements are correct:
- `memory_in_bytes` is sum of `terms_*`, `stored_fields_*`, `norms_*`, `term_vectors_*` and `doc_values_*` (and probably also `points_*`, but it is not added here anyway). 
- the rest is independent from each other
So, I'll add the following metrics: `elasticsearch.index.segments.memory` with attributes, `elasticsearch.index.segments.size`, `elasticsearch.index.segments.count`.